pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -150,3 +235,54 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	fmt.Fprintf(out, "%s\n", newName)
 	return nil
 }
+
+func hashObject(obj runtime.Object, codec runtime.Codec) (string, error) {
+	data, err := codec.Encode(obj)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%x", md5.Sum(data)), nil
+}
+
+func addDeploymentKeyToReplicationController(oldRc *api.ReplicationController, client *client.Client, deploymentKey, namespace string) error {
+	oldHash, err := hashObject(oldRc, client.Codec)
+	if err != nil {
+		return err
+	}
+	// Update all labels to include the new hash, so they are correctly adopted
+	podList, err := client.Pods(namespace).List(labels.SelectorFromSet(oldRc.Spec.Selector), fields.Everything())
```

Comment:
Ideally, this would use the label update command, which we'd factor out into a library package, at least for the pod updates. We also need to update the selector on the RC, but there's not currently a generic way to do that (though I suspect we'll need one), so I'm more sympathetic to custom code for that.